core/trino-main/src/main/java/io/trino/operator/DeduplicatingDirectExchangeBuffer.java

```diff
@@ -153,30 +189,25 @@ public synchronized void addPages(TaskId taskId, List<Slice> pages)
             return;
         }
 
-        checkState(allTasks.contains(taskId), "task is not registered: %s", taskId);
-        checkState(!successfulTasks.contains(taskId), "task is finished: %s", taskId);
-        checkState(!failedTasks.containsKey(taskId), "task is failed: %s", taskId);
-
         if (failure != null) {
             return;
         }
 
+        checkState(allTasks.contains(taskId), "task is not registered: %s", taskId);
+        checkState(!successfulTasks.contains(taskId), "task is finished: %s", taskId);
+        checkState(!failedTasks.containsKey(taskId), "task is failed: %s", taskId);
```

Comment:
move of this code block looks unrelated. Separate commit?

Reply:
Actually yeah, let me move it